### PR TITLE
feat: revamped cloud storage

### PR DESCRIPTION
**Pull request by Panaetius**
PRs needed for this:
- https://github.com/SwissDataScienceCenter/renku-notebooks/pull/1614
- https://github.com/SwissDataScienceCenter/renku-ui/pull/2760
- https://github.com/SwissDataScienceCenter/renku-gateway/pull/670
- https://github.com/SwissDataScienceCenter/renku-data-services/pull/52 
- https://github.com/SwissDataScienceCenter/renku-actions/pull/65 (and update the action everywhere. just so /deploy works with data services)
- renku-graph releaseoff of develop (since the UI depends on it)
- ( https://github.com/SwissDataScienceCenter/renku-python/pull/3629 technically not needed but relevant for users)
- https://github.com/SwissDataScienceCenter/renku-data-services/pull/53

/deploy #persist extra-values=dataService.image.pullPolicy=Always,global.gitlab.url=https://gitlab.dev.renku.ch,notebooks.cloudstorage.s3.enabled=true,amalthea.extraChildResources[0].group=com.ie.ibm.hpsys,amalthea.extraChildResources[0].name=datasets,notebooks.cloudstorage.s3.installDatashim=false

### CODE REVIEW

**Comment by olevski** [comment]
@Panaetius I know this is still a WIP. But this contains breaking changes to the values file of the helm chart. So please before merging this make sure you mention the Helm chart values changes both in `CHANGELOG.rst` and in `helm-chart/values.yaml.changelog.md`.

File: helm-chart/renku/templates/notebooks/test.yaml [hunk 1]
```diff
@@ -113,8 +113,8 @@ spec:
         - name: NB_SESSIONS__SSH__HOST_KEY_SECRET
           value: {{ $.Values.notebooks.ssh.hostKeySecret | quote }}
         {{- end }}
-        - name: NB_CRC_URL
-          value: {{ printf "http://%s/api/data" $.Values.global.crc.serviceName }}
+        - name: NB_DATA_SERVICE_URL
+          value: {{ printf "http://%s/api/data" $.Values.global.dataService.serviceName }}
```
Comment (olevski) [inline helm-chart/renku/templates/notebooks/test.yaml:117]: You fully removed `.globa.dataService.serviceName` from the values. So better to use the same logic of putting the name together as in notebooks. Or even better make a template for it in `_helpers.tpl` and use the same value everywhere.

**Comment by olevski** [comment]
@lorenzo-cavazzi this is missing the ui release. Everything else is ready to go. Please commit the ui release update directly in this PR.

**Comment by olevski** [comment]
The tests are failing because they timeout when they wait for the sessions to be deleted. So I increased the timeout in this case from `TIMEOUTS.long` to `TIMEOUTS.vlong` - which is from 1min to 3mins.

**Review by olevski — APPROVED** [review]

**Review by bethcg — APPROVED** [review]

**Pull request merged**